internal/build/phases.go

```diff
@@ -30,13 +41,24 @@ func (l *Lifecycle) Detect(ctx context.Context, networkMode string, volumes []st
 	if err != nil {
 		return err
 	}
+
+	detect, err := phaseFactory.New(
+		phaseName,
+		configProvider,
+	)
+	if err != nil {
+		return err
+	}
+
 	defer detect.Cleanup()
 	return detect.Run(ctx)
 }
 
-func (l *Lifecycle) Restore(ctx context.Context, cacheName string) error {
-	restore, err := l.NewPhase(
-		"restorer",
+func (l *Lifecycle) Restore(ctx context.Context, cacheName string, phaseFactory PhaseFactory) error {
+	phaseName := "restorer"
+
+	configProvider, err := NewPhaseConfigProvider(
+		phaseName,
```

Comment:
This is a little awkward; maybe the config provider can hold onto the name and expose it to the factory?